lib/postprocessing/image.js

```diff
@@ -16,10 +16,163 @@ const gm = require("./gm-promisify");
 const util = require('util');
 const exec = util.promisify(require('child_process').exec);
 
+// list of supported output formats
+// maps our fmt ('png') to imagemagick format ('PNG') https://imagemagick.org/script/formats.php
+const SUPPORTED_OUTPUT_FMT = {
+    "png": "PNG",
+    "jpg": "JPEG",
+    "jpeg": "JPEG",
+    "tif": "TIFF",
+    "tiff": "TIFF",
+    "gif": "GIF"
+};
+
+// list of supported input formats (intermediate rendition)
+// in imagemagick format https://imagemagick.org/script/formats.php
+const SUPPORTED_INPUT_FMT = new Set([
+    "PNG",
+    "JPEG",
+    "JPEG",
+    "TIFF",
+    "TIFF",
+    "GIF",
+    "BMP"
+]);
+
+// minimal instructions which many workers can already handle themselves
+// const BASIC_INSTRUCTIONS = new Set([
+//     "fmt",
+//     "width",
+//     "height",
+//     "quality"
+// ]);
+
+// list of post processing instructions we handle
+const EXTRA_INSTRUCTIONS = new Set([
+    "crop",
+    "jpegSize",
+    "interlace",
+    "dpi",
+    "convertToDpi"
+]);
+
+function isSupportedOutputFormat(instructions) {
+    return Object.keys(SUPPORTED_OUTPUT_FMT).includes(instructions.fmt);
```

Comment:
`SUPPORTED_OUTPUT_FMT.has(nstructions.fmt);` will do the same and use the Set API.

Reply:
`SUPPORTED_OUTPUT_FMT` is an object, not a set. We need both our `fmt` strings and the imagemagick format strings, that's why it's an object.